app/views/auth.py

```diff
@@ -0,0 +1,121 @@
+from flask.views import MethodView
+from flask import jsonify, request
+from ..models import db, User, Bucketlist, Activity
+from . import auth_blueprint
+
+
+class RegistrationApiView(MethodView):
+    """
+        class that handles all authentications
+    """
+
+    def post(self):
+        """function that registers the user"""
+
+        user_data = request.get_json()
+        print("r  ", user_data)
+        user = User.query.filter_by(email=user_data.get('email')).first()
+        if not user:
+            try:
+                user = User(username=user_data.get('username'),
+                            email=user_data.get('email'),
+                            password=user_data.get('password'))
+            user = user
+                # saves user
+                user.save()
+                responseObject= {
+                    'status': 'success',
+                    'message': 'You were successfully registered',
+                }
+                return jsonify(responseObject), 201
+            except Exception as e:
+                responseObject = {
+                    'message': str(e) 
+                }
+                return jsonify(responseObject), 500
+        else:
+            responseObject = {
+                'status': 'fail',
+                'message': 'User already exist please Log in.'
+            }
+            return jsonify(responseObject), 409
+
+class LoginAPI(MethodView):
+    """
+        class that handles User Login
+    """
+    
+    def post(self):
+        """Function that handles user login"""
+        
+        user_data = request.get_json()
+        user = User.query.filter_by(email=user_data.get('email')).first()
+        g.user
+        if user and user.verify_password(user_data.get('password')):
+            try:
+                access_token = user.generate_token(user.id)
+                if access_token:
+                    responseObject = {
+                        'status': 'success',
+                        'message': 'You are logged in!',
+                        'access_token': access_token.decode()
+                    }
+                    return jsonify(responseObject), 200
+
+            except Exception as e:
+                responseObject = {
+                    'status': 'fail',
+                    'message': str(e)
+                }
+                return jsonify(responseObject),  400
```

Comment:
Status code `400` is for bad request, I do not think it's appropriate to use it for an exception.
Check out this[http://www.restapitutorial.com/httpstatuscodes.html](url)